ur_robot_driver/test/integration_test.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+# Copyright 2019, FZI Forschungszentrum Informatik
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import os
+import time
+import pytest
+
+import launch_testing
+from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument, IncludeLaunchDescription
+from launch.substitutions import LaunchConfiguration
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+
+import rclpy
+from rclpy.node import Node
+
+from ur_msgs.srv import SetIO
+from ur_msgs.msg import IOStates
+from ur_dashboard_msgs.srv import GetRobotMode
+from ur_dashboard_msgs.msg import RobotMode
+from std_srvs.srv import Trigger
+
+
+@pytest.mark.launch_test
+def generate_test_description():
+    declared_arguments = []
+
+    declared_arguments.append(
+        DeclareLaunchArgument(
+            "robot_ip",
+            default_value="192.168.56.101",
+            description="IP address by which the robot can be reached.",
+        )
+    )
+
+    declared_arguments.append(
+        DeclareLaunchArgument(
+            "ur_type", default_value="ur5e", description="Type/series of used UR robot."
+        )
+    )
+
+    ur_type = LaunchConfiguration("ur_type")
+    robot_ip = LaunchConfiguration("robot_ip")
+    dir_path = os.path.dirname(os.path.realpath(__file__))
```

Comment:
Would it make sense to use here `ThisLaunchFileDir`?

Include: `from launch.substitutions import ThisLaunchFileDir`
Use: `ThisLaunchFileDir()` - see e.g. [here](https://github.com/ros-controls/ros2_control_demos/blob/4888549d2f8ad6de9e025bdf62ad4541b6674729/ros2_control_demo_bringup/launch/rrbot_system_position_only.launch.py#L69)